implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/v2/DynatraceExporterV2.java

```diff
@@ -0,0 +1,383 @@
+                        logger.debug(String.format(ILLEGAL_ARGUMENT_EXCEPTION_FORMATTER, meter.getId().getName(), iae.getMessage()));
+                    }
+                    return null;
+                })
+                .filter(Objects::nonNull);
+    }
+
+    Stream<String> toCounter(Meter meter) {
+        return streamOf(meter.measure()).map(
+                measurement -> {
+                    try {
+                        throwIfValueIsInvalid(measurement.getValue());
+                        return createMetricBuilder(meter)
+                                .setDoubleCounterValueDelta(measurement.getValue())
+                                .serialize();
+                    } catch (MetricException e) {
+                        logger.warn(String.format(METRIC_EXCEPTION_FORMATTER, meter.getId().getName(), e.getMessage()));
+                    } catch (IllegalArgumentException iae) {
+                        // drop lines containing NaN or Infinity silently.
+                        logger.debug(String.format(ILLEGAL_ARGUMENT_EXCEPTION_FORMATTER, meter.getId().getName(), iae.getMessage()));
+                    }
+                    return null;
+                })
+                .filter(Objects::nonNull);
+    }
+
+    private Metric.Builder createMetricBuilder(Meter meter) {
+        return metricBuilderFactory
+                .newMetricBuilder(meter.getId().getName())
+                .setDimensions(fromTags(meter.getId().getTags()))
+                .setTimestamp(Instant.ofEpochMilli(clock.wallTime()));
+    }
+
+    private static DimensionList fromTags(List<Tag> tags) {
+        return DimensionList.fromCollection(
+                tags.stream()
+                        .map(tag -> Dimension.create(tag.getKey(), tag.getValue()))
+                        .collect(Collectors.toList()));
+    }
+
+    static <T> Stream<T> streamOf(Iterable<T> iterable) {
+        return StreamSupport.stream(iterable.spliterator(), false);
+    }
+
+    private void send(List<String> metricLines) {
+        try {
+            String body = String.join("\n", metricLines);
+            if (logger.isDebugEnabled()) {
+                logger.debug("sending lines:\n" + body);
+            }
+
+            httpClient.post(endpoint)
+                    .withHeader("Authorization", "Api-Token " + config.apiToken())
+                    .withHeader("Content-Type", "text/plain")
+                    .withHeader("User-Agent", "micrometer")
+                    .withPlainText(body)
+                    .send()
+                    .onSuccess((r) -> handleSuccess(metricLines.size(), r))
+                    .onError((r) -> logger.error("Failed metric ingestion. code={} response.body={}", r.code(), r.body()));
+        } catch (Throwable throwable) {
+            logger.error("Failed metric ingestion: {}", throwable.getMessage());
+        }
+    }
+
+    private void handleSuccess(int totalSent, HttpSender.Response r) {
```

Comment:
How about `response`?
```suggestion
    private void handleSuccess(int totalSent, HttpSender.Response response) {
```